src/_base/_twig/docker-compose.yml/service/solr.yml.twig

```diff
@@ -20,13 +20,19 @@
 {% if @('services.solr.config_path') %}
       - /opt/solr/server/solr/configsets/{{ @('services.solr.environment.SOLR_CORE_NAME') }}
 {% elseif @('services.solr.major_version') == 4 %}
-      - /opt/solr/example/example-schemaless/solr/{{ @('services.solr.environment.SOLR_CORE_NAME') }}/conf
+      - /opt/solr/example/example-schemaless/solr/{{ @('services.solr.environment.SOLR_CORE_NAME') }}
 {% endif %}
     volumes:
 {% if @('services.solr.config_path') %}
-      - {{ @('services.solr.config_path')}}:/opt/solr/server/solr/configsets/{{ @('services.solr.environment.SOLR_CORE_NAME') }}
+      - {{ @('services.solr.config_path')}}:/opt/solr/server/solr/configsets/{{ @('services.solr.environment.SOLR_CORE_NAME') }}/conf
+{% endif %}
+{% if @('services.solr.major_version') == 4 %}
+      - solr_data:/opt/solr/example/solr
+{% elseif @('services.solr.major_version') < 8 %}
+      - solr_data:/opt/solr/server/solr/mycores
+{% else %}
+      - solr_data:/var/solr
 {% endif %}
-      - solr_data:/var/solr/{{ @('services.solr.environment.SOLR_CORE_NAME') }}/data/
```

Comment:
as solr_data is a named volume, this will error as there's no way to specify named volumes without the docker-compose.yml.twig also being aware of it to add a volumes section to define the named volume.

Looks like a bug in the target branch as well